components/searchable_user_list/searchable_user_list.tsx

```diff
@@ -2,46 +2,47 @@
 // See LICENSE.txt for license information.
 /* eslint-disable react/no-string-refs */
 
-import $ from 'jquery';
-import PropTypes from 'prop-types';
-import React from 'react';
-import ReactDOM from 'react-dom';
+import React, {createRef, ReactNode} from 'react';
+
 import {FormattedMessage, injectIntl} from 'react-intl';
 
 import QuickInput from 'components/quick_input';
 import UserList from 'components/user_list.jsx';
 import LocalizedInput from 'components/localized_input/localized_input';
 
 import {t} from 'utils/i18n';
+import {IntlProps} from 'components/search_results/types';
 
 const NEXT_BUTTON_TIMEOUT = 500;
 
-class SearchableUserList extends React.PureComponent {
-    static propTypes = {
-        users: PropTypes.arrayOf(PropTypes.object),
-        usersPerPage: PropTypes.number,
-        total: PropTypes.number,
-        extraInfo: PropTypes.object,
-        nextPage: PropTypes.func.isRequired,
-        previousPage: PropTypes.func.isRequired,
-        search: PropTypes.func.isRequired,
-        actions: PropTypes.arrayOf(PropTypes.func),
-        actionProps: PropTypes.object,
-        actionUserProps: PropTypes.object,
-        focusOnMount: PropTypes.bool,
-        renderCount: PropTypes.func,
-        filter: PropTypes.string,
-        renderFilterRow: PropTypes.func,
-        page: PropTypes.number.isRequired,
-        term: PropTypes.string.isRequired,
-        onTermChange: PropTypes.func.isRequired,
-        intl: PropTypes.any,
-        isDisabled: PropTypes.bool,
-
-        // the type of user list row to render
-        rowComponentType: PropTypes.func,
-    };
-
+ type Props = {
+     users: Array<Record<string, unknown>>;
+     usersPerPage: number;
+     total: number;
+     extraInfo?: Record<string, unknown>;
+     nextPage: (page?: number) => Promise<void> | void;
+     previousPage: (page?: number) => Promise<void> | void;
+     search: (term: string) => void;
+     actions?: ReactNode[];
```

Comment:
I think the correct type for this is `React.ElementType`, not `ReactNode`. I'm not 100% sure since `UserList` isn't migrated, but that seems correct.

The difference between the two is that `ElementType` is used when you're passing around a component class (or functional component or HTML element name), and `ReactNode` is the result of rendering something.
```js
let Element: React.ElementType;
let component: React.ReactNode = <Element/>;
```

Reply:
@greenbackboogie97 Can we address this as well?